lib/ood_core/job/adapters/lsf/batch.rb

```diff
@@ -0,0 +1,156 @@
+  # @param bin [#to_s] path to LSF installation binaries
+  def initialize(bin: "")
+    # TODO: @cluster = cluster.to_s
+    @bin     = Pathname.new(bin.to_s)
+  end
+
+  # Get a list of hashes detailing each of the jobs on the batch server
+  # @param id [#to_s] the id of the job to check (if just checking one job)
+  # @param filters [Array<Symbol>] list of attributes to filter on
+  # @raise [Error] if `bjobs` command exited unsuccessfully
+  # @return [Array<Hash>] list of details for jobs
+  def get_jobs(id: "", filters: [])
+    id = id.to_s
+    #TODO: does filters make sense here? regardless this is a speed enhancement
+    # selected_fields = filters.empty? ? fields : (filters & fields)
+    #TODO: split into get_all_jobs, get_my_jobs, get_job?
+    args = bjobs_default_args
+
+    #TODO: get_jobs with a list of ids...
+    # get_jobs(ids = []) *(ids.map(&:to_s))
+    args << id unless id.empty?
+
+    parse_bjobs_output call("bjobs", *args)
+  end
+
+  def bjobs_default_args
+    %w( -u all -a -w -W )
+  end
+
+  # helper method
+  def parse_bjobs_output(response)
+    return [{}] if response =~ /No job found/ || response.nil?
+
+    lines = response.split("\n")
+    validate_bjobs_output_columns(lines.first.split)
+
+    lines.drop(1).map{ |job|
+      values = split_bjobs_output_line(job)
+
+      # make a hash of { field: "value", etc. }
+      Hash[fields.zip(values)].each_with_object({}) { |(k,v),o|
+        # if the value == "-", replace it with nil
+        o[k] = (v == "-" ? nil : v)
+      }
+    }
+  end
+
+
+  # Put a specified job on hold
+  # @example Put job "1234" on hold
+  #   my_batch.hold_job("1234")
+  # @param id [#to_s] the id of the job
+  # @raise [Error] if `bstop` command exited unsuccessfully
+  # @return [void]
+  def hold_job(id)
+    call("bstop", id.to_s)
+  end
+
+  # Release a specified job that is on hold
+  # @example Release job "1234" from on hold
+  #   my_batch.release_job("1234")
+  # @param id [#to_s] the id of the job
+  # @raise [Error] if `bresume` command exited unsuccessfully
+  # @return [void]
+  def release_job(id)
+    call("bresume", id.to_s)
+  end
+
+  # Delete a specified job from batch server
+  # @example Delete job "1234"
+  #   my_batch.delete_job("1234")
+  # @param id [#to_s] the id of the job
+  # @raise [Error] if `bkill` command exited unsuccessfully
+  # @return [void]
+  def delete_job(id)
+    call("bkill", id.to_s)
+  end
+
+  # Submit a script expanded as a string to the batch server
+  # @param str [#to_s] script as a string
+  # @param args [Array<#to_s>] arguments passed to `sbatch` command
+  # @param env [Hash{#to_s => #to_s}] environment variables set
+  # @raise [Error] if `bsub` command exited unsuccessfully
+  # @return [String] the id of the job that was created
+  def submit_string(str, args: [], env: {})
+    args = args.map(&:to_s)
+    parse_bsub_output call("bsub", *args, env: env, stdin: str.to_s)
+  end
+
+  # helper method
```

Comment:
Its the batch object, not the adapter. I'd like to have the helpers be public. Its easier to test without mocks. It might make more sense though to move this to a batch_helper.rb class.